openstack/intel-worker/tasks/update-chrony-configuration.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: Update the Chrony config
+  hosts: all
+  vars_files:
+    - ../vars/main.yml
+
+  tasks:
+
+  - name: Generate the subnet list
+    shell:
+      cmd: |
+        # Create the subnets.yml
+        cat << EOF > subnets.yml
+        ---
+        subnets:
+        EOF
+        # Populate data in subnets.yml
+        for SUBNET in $(ip r | grep -v default | awk '{print $1}')
+        do
+        cat << EOF >> subnets.yml
+          - { subnet: '${SUBNET}'}
+        EOF
+        done
+
+  - name: Backup the prior Chrony configuration
+    shell:
+      cmd: |
+        mv /etc/chrony.conf.backup /etc/chrony.conf.backup-$(date +%s) || true
+        cp -f /etc/chrony.conf /etc/chrony.conf.backup
+
+  - name: Include vars of the subnets.yml
+    include_vars: 
+      file: subnets.yml
+
+  - name: Update the chrony config
+    ansible.builtin.replace:
+      path: /etc/chrony.conf
+      regexp: "# Allow NTP client access from local network.\n"
+      replace: "# Allow NTP client access from local network.\nallow {{ item.subnet }}\n"
+    loop: "{{ subnets }}"
+  
```

Comment:
Create a new file update-chrony-subnet.yml with the below... it'll make the changes idempotent

```
---
- name: Conditionally update allow subnet for chrony.conf
  shell:
    grep '^allow {{item}}$' /etc/chrony.conf
  register: result
  failed_when: result.rc != 0 and result.rc != 1
- name: Reporting if this is matched
  ansible.builtin.debug:
    var: result.rc
- name: Update the chrony config
  ansible.builtin.replace:
    path: /etc/chrony.conf
    regexp: "# Allow NTP client access from local network.\n"
    replace: "# Allow NTP client access from local network.\nallow {{ item }}\n"
  when: result.stdout == ""
```